crates/rolldown/src/ecmascript/ecma_module_factory.rs

```diff
@@ -73,7 +74,7 @@ impl ModuleFactory for EcmaModuleFactory {
       &stable_id,
       ctx.options,
       &ctx.module_type,
-      args.source,
+      args.source.clone(),
```

Comment:
Clone the source maybe not a good idea, but you could merge it first. Maybe this is the  reason performance regressed in #2251

Reply:
Yeah. Let's merge it first. I have a way to solve it in later PRs.